### ISSUE TITLE
Shouldn't the removal of `__TestExhaustive` require a version bump?

### ISSUE THREAD

**Issue by setzer22** [dtolnay/syn#1152]
*I'm afraid I must insist, so let me try again, and if you feel the need to close this one too then rest assured I won't bother you again*

I must say I'm very surprised about the reaction to #1151. If I came across as offensive in any way, please accept my apologies because this was the furthest from my intention.

I admit I was dealing with some code that was, inadvertently, matching against the `__TestExhaustive` enum variant, and having now read the documentation more carefully, I understand what was the intended usage of that variant: That is, to break test builds when something changes.

That being said, I still want to ask you to reconsider your position: Just like me, many others might've been using their IDEs to automatically fill all the enum variants, and I hope you agree with me that the docstring in a `#[doc(hidden)]` variant in enums with so many variants is very easy to miss and was not a very good place to document something that may affect the satbility of future builds.

**Comment by dtolnay** [comment]
IDEs do all variety of nutty things. If the IDE is filling your source code with doc(hidden) stuff, and double underscores which are conventionally strongly suggestive of non-public APIs in Rust, then that is between you and the IDE that you choose to use.

Sorry about locking your issues promptly &mdash; similar ones have occasionally gotten brigaded, and since I'm not planning to undo this particular change I'd prefer to prevent that.
